src/hibernate-resume/hibernate-resume.c

```diff
@@ -1,50 +1,58 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include <errno.h>
-#include <stdio.h>
 #include <sys/stat.h>
 
-#include "alloc-util.h"
 #include "devnum-util.h"
 #include "fileio.h"
 #include "initrd-util.h"
 #include "log.h"
+#include "main-func.h"
+#include "parse-util.h"
+#include "static-destruct.h"
+#include "string-util.h"
 
-int main(int argc, char *argv[]) {
+static char *arg_resume_device = NULL;
+static uint64_t arg_resume_offset = 0;
+
+STATIC_DESTRUCTOR_REGISTER(arg_resume_device, unsetp);
+
+static int run(int argc, char *argv[]) {
         struct stat st;
-        const char *device;
         int r;
 
-        if (argc != 2) {
-                log_error("This program expects one argument.");
-                return EXIT_FAILURE;
-        }
-
         log_setup();
 
+        if (argc != 3)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "This program expects two arguments.");
+
         umask(0022);
 
-        /* Refuse to run unless we are in an initrd() */
         if (!in_initrd())
-                return EXIT_SUCCESS;
+                return 0;
 
-        device = argv[1];
+        arg_resume_device = argv[1];
 
-        if (stat(device, &st) < 0) {
-                log_error_errno(errno, "Failed to stat '%s': %m", device);
-                return EXIT_FAILURE;
-        }
+        r = safe_atou64(argv[2], &arg_resume_offset);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse resume offset %s: %m", argv[2]);
 
-        if (!S_ISBLK(st.st_mode)) {
-                log_error("Resume device '%s' is not a block device.", device);
-                return EXIT_FAILURE;
-        }
+        if (stat(arg_resume_device, &st) < 0)
+                return log_error_errno(errno, "Failed to stat resume device '%s': %m", arg_resume_device);
+
+        if (!S_ISBLK(st.st_mode))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Resume device '%s' is not a block device.", arg_resume_device);
+
+        r = write_string_filef("/sys/power/resume_offset", WRITE_STRING_FILE_DISABLE_BUFFER, "%" PRIu64, arg_resume_offset);
+        if (r < 0)
+                return log_error_errno(r,
+                                       "Failed to write resume offset %" PRIu64 " to /sys/power/resume_offset for device '%s': %m",
+                                       arg_resume_offset, arg_resume_device);
```

Comment:
hmm, this sysfs was added late iirc. maybe if we get ENOENT here and the offset is zero just ignore the failure (but still debug log)